joomla/joomla-cms
# Use simpler db query for admin mod_menu, without GROUP BY

Pull Request for Issue #12983 and in general.

### Summary of Changes
Use more simpler and faster query and return the same result.

* return the same result
* do not use GROUP BY

### Testing Instructions

1. Go to backend and check whether menu items display the same list of items before and after patch.
2. It would be good to test it on multi language menu.

Before query:
```sql
SELECT a.*, SUM(b.home) AS home,b.language,l.image,l.sef,l.title_native
FROM j37_menu_types AS a
LEFT JOIN j37_menu AS b ON b.menutype = a.menutype AND b.home != 0
LEFT JOIN j37_languages AS l ON l.lang_code = language
WHERE (b.client_id = 0 OR b.client_id IS NULL)
GROUP BY a.id, a.menutype, a.description, a.title, b.menutype,b.language,l.image,l.sef,l.title_native
```


![menu_no_group2](https://user-images.githubusercontent.com/9054379/27791810-2a65a85a-5ff7-11e7-8c7c-a9240e05951d.png)


After query:
```sql
SELECT a.id, a.asset_id, a.menutype, a.title, a.description, a.client_id,c.home, c.language, c.image, c.sef, c.title_native 
FROM j37_menu_types AS a 
LEFT JOIN ( 
    SELECT b.menutype, b.home, b.language, l.image, l.sef, l.title_native 
    FROM j37_menu AS b 
    LEFT JOIN j37_languages AS l ON l.lang_code = b.language 
    WHERE b.home != 0 AND (b.client_id = 0 OR b.client_id IS NULL)
) c ON c.menutype = a.menutype
ORDER BY a.id
```

![menu_no_group3](https://user-images.githubusercontent.com/9054379/27792044-2ee61e90-5ff8-11e7-9829-987fb9797f44.png)


### Expected result
Works as before but for tables with more rows it works faster.


### Documentation Changes Required
None
